### PR TITLE
Internet midi respects start and end times

### PR DESCRIPTION
**Pull request by JJRcop**
I was looking through [my rounds](https://sb.atlantaned.space/me) on statbus because I really wanted to find this dumb song that an admin played a while back, that I couldn't find anymore. I still haven't found it, but I did find this

[screenshot_20190305_131352](https://sb.atlantaned.space/rounds/82772/played_url)

So I thought "hm, I know that the admin midi definitely does not work like that", so I checked the logs of that round

[cobby_struggling_with_memes](https://tgstation13.org/parsed-logs/basil/data/logs/2018/01/30/round-82772/game.html)

Well @ExcessiveUseOfCobblestone, now it does!

Also, later on in the PR's life, I found that youtube-dl parses end_time too, so I supported that cause it's cool.

:cl: JJRcop
admin: Play Internet Sound now respects start (like &t=xxx or &start=xxx) and end times (like &end=xxx). This allows admins to choose a snippet of media to play, like for example pick a song out of an hour long album compilation.
/:cl:

### CODE REVIEW

**Comment by ExcessiveUseOfCobblestone** [comment]
lol

**Comment by ExcessiveUseOfCobblestone** [comment]
wonder what song it was

**Comment by subject217** [comment]
you can click on the "screencap" cobby it's actually a link

**Comment by ExcessiveUseOfCobblestone** [comment]
LOL fighting the fire boss with fire

File: code/modules/admin/verbs/playsound.dm [hunk 1]
```diff
@@ -67,7 +67,7 @@
 	if(istext(web_sound_input))
 		var/web_sound_url = ""
 		var/stop_web_sounds = FALSE
-		var/pitch
+		var/list/music_extra_data = list("pitch", "seek")
```
Comment (ninjanomnom) [inline code/modules/admin/verbs/playsound.dm]: Just start it off as an empty list, you can assign to nonexistent keys without issue and trying to access those same keys will return null just fine

File: code/modules/goonchat/browserOutput.dm [hunk 2]
```diff
@@ -115,12 +115,15 @@ GLOBAL_DATUM_INIT(iconCache, /savefile, new("tmp/iconCache.sav")) //Cache of ico
...
 
-/datum/chatOutput/proc/sendMusic(music, pitch)
+/datum/chatOutput/proc/sendMusic(music, list/extra_data)
 	if(!findtext(music, GLOB.is_http_protocol))
 		return
 	var/list/music_data = list("adminMusic" = url_encode(url_encode(music)))
-	if(pitch)
-		music_data["musicRate"] = pitch
+
+	if(extra_data && extra_data.len)
```
Comment (ninjanomnom) [inline code/modules/goonchat/browserOutput.dm]:
```suggestion
	if(extra_data?.len)
```

**Review by ninjanomnom — DISMISSED** [review]

File: code/modules/goonchat/browserOutput.dm (hunk 2 above)
Comment (SouDescolado) [reply code/modules/goonchat/browserOutput.dm]: @ninjanomnom This always works? Like it checks if it is not null and then checks the "subvalue"?

Comment (ExcessiveUseOfCobblestone) [reply code/modules/goonchat/browserOutput.dm]: it's equivalent

File: code/modules/admin/verbs/playsound.dm (hunk 1 above)
Comment (JJRcop) [reply code/modules/admin/verbs/playsound.dm]: Thanks, I am already aware of this. The intent is to let people know what keys are there to be used easily. With that in mind, do you still want it to be an empty list?

File: code/modules/goonchat/browserOutput.dm (hunk 2 above)
Comment (JJRcop) [reply code/modules/goonchat/browserOutput.dm]: That's cool! I'll implement that right away

File: code/modules/admin/verbs/playsound.dm (hunk 1 above)
Comment (JJRcop) [reply code/modules/admin/verbs/playsound.dm]: This issue is still valid in my most recent commit, I simply added "end" to the list.

**Comment by JJRcop** [comment]
Now it supports end time too! Normally youtube doesn't do end time on the website, it just ignores the tag, which is &end. However, youtube-dl parses it anyway and we can use it!

**Review by ninjanomnom — APPROVED** [review]

**Pull request merged**